clash-protocols/src/Protocols/Internal/TH.hs

```diff
@@ -56,8 +28,8 @@ idleCircuitTupleInstance n =
     |]
  where
   circTys = map (\i -> varT $ mkName $ "c" <> show i) [1 .. n]
-  instCtx = foldl appT (tupleT n) $ map (\ty -> [t|IdleCircuit $ty|]) circTys
-  instTy = foldl appT (tupleT n) circTys
+  instCtx = appTs (tupleT n) $ map (\ty -> [t|IdleCircuit $ty|]) circTys
+  instTy = appTs (tupleT n) circTys
```

Comment:
I think you accidentally applied this again, as we agreed `foldl appT` is easier for the reader.